tests/feature_activation/test_feature_simulation.py

```diff
@@ -0,0 +1,194 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Any
+from unittest.mock import Mock, patch
+
+from hathor.feature_activation import feature_service
+from hathor.feature_activation.feature import Feature
+from hathor.feature_activation.model.criteria import Criteria
+from hathor.feature_activation.resources.feature import FeatureResource
+from hathor.feature_activation.settings import Settings as FeatureSettings
+from hathor.simulator.trigger import StopAfterNMinedBlocks
+from tests import unittest
+from tests.resources.base_resource import StubSite
+from tests.simulation.base import SimulatorTestCase
+
+_FEATURE_SETTINGS = FeatureSettings(
+    evaluation_interval=4,
+    max_signal_bits=4,
+    default_threshold=3,
+    features={
+        Feature.NOP_FEATURE_1: Criteria(
+            bit=0,
+            start_height=20,
+            timeout_height=60,
+            activate_on_timeout=True,
+            version='0.0.0'
+        )
+    }
+)
+
+
+class BaseTestFeatureSimulation(SimulatorTestCase):
+    seed_config = 5604003716498505253
```

Comment:
Why are you fixing the seed?

Reply:
If I don't, the hash of mined blocks becomes different, and the hash is asserted in the tests

Reply:
I guess you can't count on that because future changes on the full node might have a similar impact. We've had the same issue on other tests. Check them out for ways to workaround a fixed seed:

- https://github.com/HathorNetwork/hathor-core/blob/master/tests/consensus/test_soft_voided.py
- https://github.com/HathorNetwork/hathor-core/blob/master/tests/consensus/test_soft_voided3.py
- https://github.com/HathorNetwork/hathor-core/blob/master/tests/consensus/test_soft_voided4.py

Reply:
Oh, ok. Then maybe I'll change it so hashes are not asserted, only the height (which is what is actually important to test)

Reply:
Thanks!